hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpNetworkingService.java

```diff
@@ -175,6 +186,14 @@ public synchronized void start() {
 
         networking.start();
         startAcceptor();
+
+        if (advancedNetworkingEnabled) {
```

Comment:
I need to think about this part.

Reply:
The idea is that users who don't have advanced network stats enabled shouldn't "pay" for them. It doesn't make sense to spend CPU cycles on collection of these stats, as we know that they're going to be zeros.

Reply:
Should this not always be enabled by default?

Also: the dynamic probe discovery stuff is pending.

Reply:
> Should this not always be enabled by default?

As far as I know, you need to enable advanced network config explicitly: https://docs.hazelcast.org/docs/3.12.2/manual/html-single/index.html#advanced-network-configuration

It makes sense that it's not enabled by default as advanced network config is about using separate ports for client/member/WAN/REST/memcache protocols.

Reply:
> Also: the dynamic probe discovery stuff is pending.

I'm not sure how it relates with this PR. Maybe you mean metrics redesign task? If that's the case, then you should check #15560